include/openthread/border_agent.h

```diff
@@ -89,13 +90,32 @@ typedef struct otBorderAgentCounters
 } otBorderAgentCounters;
 
 /**
- * Gets the counters of the Thread Border Agent.
+ * Represents information about a Border Agent session.
  *
- * @param[in]  aInstance  A pointer to an OpenThread instance.
+ * This structure is populated by `otBorderAgentGetNextSessionInfo()` during iteration over the list of sessions using
+ * an `otBorderAgentSessionIterator`.
  *
- * @returns A pointer to the Border Agent counters.
+ * To ensure consistent `mLifetime` calculations, the iterator's initialization time is stored within the iterator,
+ * and each session's `mLifetime` is calculated relative to this time.
  */
-const otBorderAgentCounters *otBorderAgentGetCounters(otInstance *aInstance);
+typedef struct otBorderAgentSessionInfo
+{
+    otSockAddr mPeerSockAddr;          ///< Socket address (IPv6 address and port number) of session peer.
+    bool       mConnected;             ///< Indicates whether the session is connected.
+    bool       mAccepedAsCommissioner; ///< Indicates whether the session is accepted as full commissioner.
```

Comment:
nit: how about simplifying to something like `mIsCommissioner`?

Reply:
Good idea. Changed it in new push. Also changes `mConnected` to be `mIsConnected` to be consistent. CLI command `ba sessions `output is also changed to use `"commissioner:{yes/no}"`